rust-runtime/aws-smithy-http-server-python/examples/Makefile

```diff
@@ -1,56 +1,65 @@
-OS := $(shell uname -s)
 SRC_DIR := $(shell git rev-parse --show-toplevel)
 CUR_DIR := $(shell pwd)
 GRADLE := $(SRC_DIR)/gradlew
+WHEELS := $(CUR_DIR)/wheels
 SERVER_SDK_DST := $(CUR_DIR)/pokemon-service-server-sdk
 CLIENT_SDK_DST := $(CUR_DIR)/pokemon-service-client
 SERVER_SDK_SRC := $(SRC_DIR)/codegen-server-test/python/build/smithyprojections/codegen-server-test-python/pokemon-service-server-sdk/rust-server-codegen-python
 CLIENT_SDK_SRC := $(SRC_DIR)/codegen-client-test/build/smithyprojections/codegen-client-test/pokemon-service-client/rust-codegen
 
-SHARED_LIBRARY_DST := $(CUR_DIR)/libpokemon_service_server_sdk.so
-ifeq ($(OS), Darwin)
-	DEBUG_SHARED_LIBRARY_SRC := $(SRC_DIR)/target/debug/libpokemon_service_server_sdk.dylib
-	RELEASE_SHARED_LIBRARY_SRC := $(SRC_DIR)/target/release/libpokemon_service_server_sdk.dylib
-else
-	DEBUG_SHARED_LIBRARY_SRC := $(SRC_DIR)/target/debug/libpokemon_service_server_sdk.so
-	RELEASE_SHARED_LIBRARY_SRC := $(SRC_DIR)/target/release/libpokemon_service_server_sdk.so
-endif
+HAS_MATURIN := $(shell command -v maturin 2> /dev/null)
 
 all: codegen
 
 codegen:
 	$(GRADLE) --project-dir $(SRC_DIR) -P modules='pokemon-service-server-sdk,pokemon-service-client' :codegen-client-test:assemble :codegen-server-test:python:assemble
-	mkdir -p $(SERVER_SDK_DST) $(CLIENT_SDK_DST)
+	mkdir -p $(SERVER_SDK_DST) $(CLIENT_SDK_DST) $(WHEELS)
 	cp -av $(SERVER_SDK_SRC)/* $(SERVER_SDK_DST)/
 	cp -av $(CLIENT_SDK_SRC)/* $(CLIENT_SDK_DST)/
 
-clippy: codegen
-	cargo clippy
+ensure-maturin:
+ifndef HAS_MATURIN
+    $(error "maturin is not available; please install it via 'pip install maturin' or 'cargo install maturin'")
+endif
 
-build: codegen
-	cargo build
-	ln -sf $(DEBUG_SHARED_LIBRARY_SRC) $(SHARED_LIBRARY_DST)
+# Note on `--compatibility linux`: Maturin by default uses `manylinux_x_y` but it is not supported 
+# by our current CI version (3.7.10), we can drop `--compatibility linux` when we switch to higher Python version.
+# For more detail: https://github.com/pypa/manylinux
+build-wheel: ensure-maturin
+	maturin build --manifest-path $(SERVER_SDK_DST)/Cargo.toml --out $(WHEELS) --compatibility linux
 
-py_check: build
-	mypy pokemon_service.py
+build-wheel-release: ensure-maturin
+	maturin build --manifest-path $(SERVER_SDK_DST)/Cargo.toml --out $(WHEELS) --compatibility linux --release
```

Comment:
Don't we need to check if we are on mac and add the compatibily for it?

Reply:
`compatibility` is only needed for Linux and it is ignored on non-Linux platforms:
```
      --compatibility [<compatibility>...]
          Control the platform tag on linux.

          Options are `manylinux` tags (for example `manylinux2014`/`manylinux_2_24`) or `musllinux` tags (for example `musllinux_1_2`) and `linux` for the native linux tag.

          Note that `manylinux1` and `manylinux2010` is unsupported by the rust compiler. Wheels with the native `linux` tag will be rejected by pypi, unless they are separately validated by `auditwheel`.

          The default is the lowest compatible `manylinux` tag, or plain `linux` if nothing matched

          This option is ignored on all non-linux platforms
```

Reply:
Thanks got it.